module_utils/foreman_helper.py

```diff
@@ -4,6 +4,36 @@
 import re
 import yaml
 
+from ansible.module_utils.basic import AnsibleModule
+
+
+class ForemanAnsibleModule(AnsibleModule):
+
+    def __init__(self, **kwargs):
+        custom_argument_spec = kwargs.pop('argument_spec', dict())
+        argument_spec = dict(
+            server_url=dict(required=True),
+            username=dict(required=True),
+            password=dict(required=True, no_log=True),
+            verify_ssl=dict(type='bool', default=True),
+            state=dict(choices=['present', 'absent'], default='present'),
+        )
+        argument_spec.update(custom_argument_spec)
+        kwargs['argument_spec'] = argument_spec
+        super(ForemanAnsibleModule, self).__init__(**kwargs)
+
+
+def filter_module_params(module):
+    return dict([(k, v) for (k, v) in module.params.items() if v is not None])
```

Comment:
Can we dict comprehension (move to Python 2.7+)? `{k: v for (k, v) in module.params.items() if v is not None}` is easier to read.

Reply:
I think making this py2.7+ is OK, yeah.